docs/apis/subsystems/communication/index.md

```diff
@@ -0,0 +1,244 @@
+        string $communicationroomname,
+        ?\stored_file $avatar = null,
+        ?\stdClass $instance = null,
+    )
+```
+
+This method takes the following parameters:
+
+- The name of the room as a string to be created.
+- The avatar of the room to be created, this is a stored file object.
+- The instance object of the communication API.
+
+For example, we want to create a room for a course with the course name as the room name and the course image as the avatar, we can use the below code.
+There can be other cases where information from course instance will be used by a plugin, for example, the dynamic field from matrix plugin has a form
+field named topic which sets the topic of the room, the instance object can be used to get the topic from the course instance and set it in the form field.
+Please refer to [Communication plugin](../../plugintypes/communication/index.md) documentation for more details.
+
+```php
+// first initialize the communication instance, provided the context is already initialized and the selected provider is available though a form or other means.
+$communication = \core_communication\api::load_by_instance(
+            context: $context,
+            component: 'core_course',
+            instancetype: 'coursecommunication',
+            instanceid: $course->id,
+            provider: 'communication_matrix',
+        );
+
+// Now call the create_and_configure_room() method to add an ad-hoc to create a room in the communication provider and configure it with the required settings.
+$communication->create_and_configure_room(
+                communicationroomname: $course->fullname,
+                avatar: $courseimage ?: null,
+                instance: $course,
+            );
+```
+
+#### Update a room and its associated information
+
+$communication->update_room() method is used to add an ad-hoc to update a room in the communication provider and configure it with the required settings.
+
+```php
+public function update_room(
+        ?int $active = null,
+        ?string $communicationroomname = null,
+        ?\stored_file $avatar = null,
+        ?\stdClass $instance = null,
+    )
+```
+
+This method takes the following parameters:
+
+- The active status of the room to be updated. Can be either 0 or 1.
+- The name of the room as a string to be updated.
+- The avatar of the room to be updated, this is a stored file object.
+- The instance object of the communication API.
+
+```php
+// First initialize the instance.
+$communication = \core_communication\api::load_by_instance(
+            context: $context,
+            component: 'core_course',
+            instancetype: 'coursecommunication',
+            instanceid: $course->id,
+            provider: 'communication_matrix',
+        );
+
+// Now update the room with the required settings. The instance will be used by dynamic fields feature of the plugin to allow the plugin to get the required information from the instance.
```

Comment:
This probably also needs to be reviewed (the comment should be split in different lines).